cmd/manager/scap.go

```diff
@@ -168,30 +171,71 @@ func (c *scapContentDataStream) FigureResources(profile string) error {
 		}
 	}
 
-	selected := getResourcePaths(c.dataStream, c.dataStream, effectiveProfile)
+	selected, _ := getResourcePaths(c.dataStream, c.dataStream, effectiveProfile, valuesList)
 	if len(selected) == 0 {
 		fmt.Printf("no valid checks found in profile\n")
 	}
 	found = append(found, selected...)
 	c.resources = found
+	DBG("c.resources: %v\n", c.resources)
 	return nil
 }
 
 // getPathsFromRuleWarning finds the API endpoint from in. The expected structure is:
 //
 //  <warning category="general" lang="en-US"><code class="ocp-api-endpoint">/apis/config.openshift.io/v1/oauths/cluster
 //  </code></warning>
-func getPathFromWarningXML(in *xmlquery.Node) []utils.ResourcePath {
+func getPathFromWarningXML(in *xmlquery.Node, valueList map[string]string) []utils.ResourcePath {
 	DBG("Parsing warning %s", in.OutputXML(false))
-	return utils.GetPathFromWarningXML(in)
+	return utils.GetPathFromWarningXML(in, valueList)
 }
 
 // Collect the resource paths for objects that this scan needs to obtain.
 // The profile will have a series of "selected" checks that we grab all of the path info from.
-func getResourcePaths(profileDefs *xmlquery.Node, ruleDefs *xmlquery.Node, profile string) []utils.ResourcePath {
+func getResourcePaths(profileDefs *xmlquery.Node, ruleDefs *xmlquery.Node, profile string, overrideValueList map[string]string) ([]utils.ResourcePath, map[string]string) {
```

Comment:
The original code consists of following steps.

1. if tailored profile exists, process it
    1. extract resource paths from `warning` elements
    2. extract extended profile (as `effectiveProfile`)
2. process profile (or extended profile)
    1. extract resource paths from `warning` elements

I want to replace not only variables in `warning` in the tailored profile (step 1.1) but also those in `warning` in the extended profile (step 2.1).  In step 2.1, we need to override values with the ones declared in the tailored profile.  It is achieved by using the second return value of `getResourcePaths` at step 1.2 for the final parameter of `getResourcePaths` at step 2.1.

Reply:
Thanks for the explanation. I think the code looks good to me, just need to remove some of `fmt.Println(err)`

Reply:
sorry I didn't pushed the latest code to the repo. Now most of the `fmt.Println()` and `fmt.Printf()` occurrences are replaced with `DBG()`.
(There remains several `fmt.PrintXX()` in `scap.go`, but they are not added by me; I think that they should also be replaced with `DBG()`, but  such a replacement would be done on another PR if needed)

One thing I'd like to discuss is the error handling In `parse_arf_result.go`. I can't find any error output in it. Instead, it ignores error or returns error to callers ([example](https://github.com/tmishina/compliance-operator/blob/custom-api-resource/pkg/utils/parse_arf_result.go#L407)). What is the best way to handle the error at [calling RenderValues](https://github.com/tmishina/compliance-operator/blob/custom-api-resource/pkg/utils/parse_arf_result.go#L95)?  Currently, I use `fmt.Printf` here.

Reply:
@Vincent056 Is there any progress on this PR? or is there anything that I need to do? thanks.

Reply:
@tmishina Sorry for the late review! I think the PR looks good, for the question regarding how to handle error output, does it make sense if we use DBG() for `RenderValues` as well? I am going to think about the best approach for CaC PR.

Reply:
@Vincent056 `DBG()` is defined in `main` package (`cmd/manager/log.go\), and is not defined in `utils` package, to which `parse_arf_result.go` belongs.

I saw several functions of `utils` package, and I imagine that the authors of `utils` functions intentionally avoid handling error in `utils` functions.  Instead, these functions just return error messages their callers, so do functions in `parse_arf_result.go`. That's why I'm wondering what is the best way to handle errors in `GetPathFromWarningXML()` in `parse_arf_result.go`.  Can `GetPathFromWarningXML()` return error when replacement fails?

Reply:
now `utils.utils.GetPathFromWarningXML()` in `par_arf_result.go`  just returns error messages, and its caller, `getPathFromWarningXML()` in `scap.go`, writes log message.

Reply:
Hi, @tmishina I just realized we are using a different way to log in to the example I gave above, I think DEBUG() and LOG() you had in there is good. s/remote/remove just nits on one of the commit message.

Reply:
@Vincent056 thank you for pointing out my typo...I've just fixed it and force-pushed the branch.

It would be better to squash commits into reasonable numbers by `git rebase -i`.  I will do that if the PR becomes ready to be merged.

Reply:
Thanks, I think the PR looks good overall.

Reply:
Thank you! What is the remaining actions for me? `git rebase master` must be done, and what else?
The test bound with this PR is still `Some checks haven’t completed yet`; is there anything I can do to solve this? (I hope `git rebase master` and force pushing a new commit can fix this...)

Reply:
@Vincent056 done `git rebase master`, and `make test-unit` finishes without error.

Reply:
I am wondering if we want to squash the commit?

Reply:
@Vincent056 I squashed the commits of this PR into one, and `rebase master` again. Now this PR is in the next of commit fc7a2579